solidity/dashboard/src/components/WithdrawETHModal.jsx

```diff
@@ -0,0 +1,157 @@
+          backgroundColor: `${colors.grey10}`,
+        }}
+      >
+        {availableETH}
+      </span>
+      <span style={{ color: `${colors.grey60}` }}>&nbsp;ETH</span>
+    </>
+  )
+})
+
+const WithdrawETHModal = ({ operatorAddress, availableETH, closeModal }) => {
+  const web3Context = useWeb3Context()
+  const { yourAddress, web3 } = web3Context
+  const showMessage = useShowMessage()
+
+  const onSubmit = useCallback(
+    async (formValues, onTransactionHashCallback) => {
+      const { ethAmount: ethToWithdraw } = formValues
+      try {
+        await tbtcAuthorizationService.withdrawUnbondedEth(
+          web3Context,
+          { operatorAddress, ethToWithdraw },
+          onTransactionHashCallback
+        )
+        showMessage({
+          type: messageType.SUCCESS,
+          title: "Success",
+          content: "Withdrawal of ETH successfully completed",
+        })
+      } catch (error) {
+        showMessage({
+          type: messageType.ERROR,
+          title: "Withdrawal of ETH has failed ",
+          content: error.message,
+        })
+        throw error
+      }
+    },
+    [operatorAddress, showMessage, web3Context]
+  )
+
+  return (
+    <>
+      <h4 style={{ marginBottom: "0.5rem" }}>Amount available to withdraw.</h4>
+      <div className="mt-1">
+        <AvailableEthCell availableETH={availableETH} />
+      </div>
+      <div className="text-validation mb-1 mt-2 flex row center">
+        <Icons.Diamond />
+        <span className="pl-1">
+          Withdrawn ETH will go the beneficiary address.
+        </span>
+      </div>
+      <WithdrawETHFormik
+        web3={web3}
+        onSubmit={onSubmit}
+        yourAddress={yourAddress}
+        closeModal={closeModal}
+      />
+    </>
+  )
+}
+
+export default React.memo(WithdrawETHModal)
+
+const WithdrawETHForm = ({ onSubmit, closeModal, ...formikProps }) => {
+  const onSubmitBtn = useCustomOnSubmitFormik(onSubmit)
+
+  return (
+    <form>
+      <FormInput
+        name="ethAmount"
+        type="text"
+        label="ETH Amount"
+        placeholder="0"
+      />
+      <div
+        className="flex row center mt-2"
+        style={{
+          borderTop: `1px solid ${colors.grey20}`,
+          margin: "0 -2rem",
+          padding: "2rem 2rem 0",
+        }}
+      >
+        <SubmitButton
+          className="btn btn-primary"
+          type="submit"
+          onSubmitAction={onSubmitBtn}
+          withMessageActionIsPending={false}
+          triggerManuallyFetch={true}
+          disabled={!formikProps.dirty}
+        >
+          withdraw eth
+        </SubmitButton>
+        <span onClick={closeModal} className="ml-1 text-link">
+          Cancel
+        </span>
+      </div>
+    </form>
+  )
+}
+const WithdrawETHFormik = withFormik({
+  validateOnChange: false,
+  validateOnBlur: false,
+  mapPropsToValues: () => ({
+    ethAmount: "0",
+  }),
+  validate: (values, { yourAddress, web3 }) => {
```

Comment:
Async validation is no needed here. We should validate if the typed amount is not greater than the current unbonded value instead of comparing the current eth balance.

```
if (!ethAmount) {
 errors.ethAmount = "Required"
} else if (valueInWei.gt(unbondedValue)) {
 errors.ethAmount = `The value should be less than ${web3Utils.fromWei(ethBalance.toString(),"ether")}`
} else if (valueInWei.lte(web3Utils.toBN(0))) {
 errors.ethAmount = "The value should be greater than 0"
}
```

Reply:
yup, good point. Changed validation for withdrawal.